src/ints/bios_vhd.cpp

```diff
@@ -127,13 +127,20 @@ imageDiskVHD::ErrorCodes imageDiskVHD::Open(const char* fileName, const bool rea
         uint8_t buf[512];
         if(fseeko64(file, 0, SEEK_SET)) { fclose(file); return INVALID_DATA; }
         if(fread(&buf, sizeof(uint8_t), 512, file) != 512) { fclose(file); return INVALID_DATA; }
+        //detect real DOS geometry (MBR, BPB or default X-255-63)
         imageDiskVHD::scanMBR(buf, sizes, footer.currentSize);
         vhd->cylinders = sizes[3];
         vhd->heads = sizes[2];
         vhd->sectors = sizes[1];
         vhd->fixedDisk = new imageDisk(file, fileName, vhd->cylinders, vhd->heads, vhd->sectors, 512, true);
+        //now possible FAT geometry can be detected (in Windows 11, those two can differ!)
+        vhd->DetectGeometry(sizes);
+        vhd->cylinders = sizes[3];
+        vhd->heads = sizes[2];
+        vhd->sectors = sizes[1];
+        vhd->fixedDisk = new imageDisk(file, fileName, vhd->cylinders, vhd->heads, vhd->sectors, 512, true);
```

Comment:
Does calling `new imageDisk()`  twice safely deletes the previous one?
I think better to check sizes[] and DetectGeometry before `new imageDisk`

Reply:
Obviously, it does not (deleting caused an obscure exception).

DetectGeometry actually requires a valid disk image internally to properly read VHD sectors, that's the reason why I avoided that call in previous commits (having tought at least Windows 1x set MBR CHS == BPB CHS).

Perhaps I'll find another way, if you haven't suggestions.